### DISCUSSION TITLE
Ability to Generate User-Customized Reports

### DISCUSSION THREAD

**Discussion by seamaiden50** [Klocman/Bulk-Crap-Uninstaller#564] [Ideas]
BC Uninstaller is one program I cannot live without, but it's missing an important feature!

It would save tons of time if the program could generate a custom report of installed software at the click of a button.

Such a button could appear at the top of the window. The default report would be based on the filtering options checked by the user on the left side of the window, as well as the settings the user has specified in Tools >> Settings.

The user should be able to customize which columns will be displayed in the report by checking/unchecking the column names in a prompt that asks "Which columns should be included in this report?". Here are some of the columns that would be helpful to select.

- Name
- Publisher
- Version
- Install Date
- Size

Depending on the size of the data in each column, the text should wrap to the next line rather than be cut off (truncated), so that all the data in each specified column will display completely. In the prompt where the user selects columns for the report, please also add options to specify if the report should be in Portrait or Landscape, and allow the user to adjust margins and printer settings. Thank you.

**Comment by Klocman** [comment]
Why not export to xml, then open in excel and pick which columns you need? You could also set up a script to automatically turn the xml export into whatever format you need.

This seems like an incredibly niche feature to implement into BCU itself when it already has support for full export, even in the CLI version.

**Comment by seamaiden50** [comment]
I didn't know it had an export to xml option. Will look into that. Thanks.

Jane Ocean
Fresno, CA, USA
***@***.***
(Pacific Time Zone)
